### DISCUSSION TITLE
File "webapp.py", line 34, in <module>

### DISCUSSION THREAD

**Discussion by kabni** [searx/searx#2648] [Installation issue: custom]
hi can some one help me with this. 

~/searx/searx │ master  python3 webapp.py                                                                                                                                                                     ✔ │ 20:00:07
Traceback (most recent call last):
  File "webapp.py", line 34, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
 ~/searx/searx │ master  ~                                                                                                                                                                                   1 х │ 20:00:25
 ~  pip3 install requests                                                                                                                                                                                      ✔ │ 20:29:24
Requirement already satisfied: requests in /usr/local/lib/python3.9/site-packages (2.25.1)
Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/lib/python3.9/site-packages (from requests) (3.0.4)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.9/site-packages (from requests) (2020.12.5)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.9/site-packages (from requests) (1.25.11)
Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.9/site-packages (from requests) (2.10)

**Comment by return42** [comment]
How did you installed searx? Most installations run in a virtualenv especially when you use our installation instructions. With other words: It seems you did not install the requirements in the virtualenv in which searx  runs (the C&P you gave is not clear for me). With your pip from above you install the packages in your home folder.

We recommend to use a docker installation or use our installation instruction, both is described [here](https://searx.github.io/searx/admin/installation.html)

**Comment by kabni** [comment]
thank you , apologies. I tried to follow the instructions - couldn't get it to work on my Mac - so I was trying to follow this: 

Step 1 : Lets install SearX
https://github.com/asciimoo/searx​

Step 2 : Now we have to go inside the SearX Directory.
cd searx

Step 3 : Let list files inside the directory.
ls

Step 4 : Now we have to install the requirements.txt file inside the Searx Directory.
pip3 install -r requirements.txt

Step 5 : Now again list files in the Directory.
ls

Step 6 : Now we have to into the searx directory.
cd searx

Step 7 : Now list files inside the Directory.
ls

Step 8 : Now we have to start the webapp.py file.
python3 webapp.py

**Comment by return42** [comment]
> couldn't get it to work on my Mac

macOS is different .. our installation instructions will not help on macOS .. we did not tested an installation on macOS / we do not have experience on this .. the installation procedure you described will not work .. sorry, that I can't help you with installing on macOS.  May you want to go to this [discussion](https://github.com/searx/searx/discussions/2629) (and close this issue)?

**Comment by kabni** [comment]
thanks sir.

**Reply by cyclaero** [reply]
For installation on macOS see my BLog post in German language:
https://obsigna.com/articles/1614714756.html#searx_for_macos

In English by courtesy of the Google translator:
https://translate.google.com/translate?hl=&sl=de&tl=en&u=https%3A%2F%2Fobsigna.com%2Farticles%2F1614714756.html%23searx_for_macos

Using my script, searx is up and running on localhost of macOS in less than 5 min.

See also my announcement on this discussion board from a few days ago: https://github.com/searx/searx/discussions/2629

**Comment by kabni** [comment]
Lovely ! Thank you sir!! 

⁣Best,
Nivelle​
